enhancements/machine-config/alignment-with-image-mode-rhel-for-the-mco.md

```diff
@@ -0,0 +1,205 @@
+
+## Motivation
+
+In older RHCOS versions we have a “non-layered” update path, then there is the rpm-ostree path. Now with Image Mode RHEL, there is a strong call for a full image-based update path. Partnering with CoreOS, MCO is also committing to its own mission to ensure OCP uses image mode and can benefit from image mode. A key advantage of converging to image mode is that layered OS images will become the source of truth for configuration that the nodes are booted to, rolled back to, and applied with. This will efficiently prevent nodes from degrading due to config drift. Keeping changes into separate image layers will also make image inspection, testing, and scanning easier. We can foresee that, down the journey, there will be more room for customization as well through bootc’s configmap support.
+
+### User Stories
+
+* As an OpenShift cluster administrator I would like MCO to leverage RHEL disk images to boot/update my nodes
+* As an Openshift administrator, I would like to reset a config I have applied without manual deletion that may cause node degradation, but via an image rollback.
+* As an OpenShift cluster administrator, I would like my cluster to understand and to be able to (automatically) target a new Openshift/RHCOS container image reference to boot for upgrade.
+* As an Openshift support engineer, I would like to easily reproduce the settings and issues my client runs into by booting according to their image queue. 
+
+### Goals
+
+* Future Day-2 configuration owned by the MCO will, by default, involve building container images from config specification and layer the image by bootc switch. 
+* Enable the user to upgrade their cluster to a rpm-ostree-based image-mode cluster and to a bootc-based image-mode cluster seamlessly 
+
+### Non-Goals
+
+* On Cluster Layering [Discussed in detail in https://github.com/openshift/enhancements/pull/1515]
+* Bootc Day-2 Configuration Tools 
+    - Description: Bootc has opened a door for disk image customization via configmap. Switching from rpm-ostree to bootc, MCO should not only make sure all the functionality remains but also proactively extending its support to make sure all the customization power brought in by bootc are directed to the user, allowing them to fully maximize these advantages.This will involve creating a new user-side API for fetching admin-defined configuration and pairing MCO with a bootc day-2 configuration tool for applying the customizations. Interest and Proposals in which has been raised several times in previous discussions.
+    - Why non-goal: This is a valid request and will enhance user experience by increasing customization option.This is a non-goal for this enhancement because this request should be the next step after bootc switch happening. It will be announced and discussed in detail in a follow-up enhancement. 
```

Comment:
This seems to be going into the weeds a bit too much. I think I would summarize it as:

```suggestion
* Bootc Day-2 Configuration Tools: new bootc features like the [planned configmap support](https://github.com/containers/bootc/issues/22) are of interest to the MCO but not in scope in this initial work to align with image-mode for RHEL.
```

Reply:
Fixed